File: frontend/src/app/dto/Product.ts
import { Brand } from "./Brand";
import { Category } from "./Category";
import { User } from "./User";

export class Product {

  id: number | null = null;
  name: string | null = null;
  code: string | null = null;
  price: string | null = null;
  category: Category | null = null;
  brand: Brand | null = null;
  addedBy: User | null = null;
  description: string | null = null;

  constructor(o?: Partial<Product>) {
    Object.assign(this, o);
  }
}

File: backend/src/main/java/com/example/backend/model/Product.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.apache.catalina.User;

import java.util.List;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String code;
    private int price;

    private String description;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY)
    private List<OrderDetail> orderDetailList;

    public Product(Long id) {
        this.id = id;
    }
}



// id ,  name
// 1     apple
// 2     orange
// 3     mango
// 4     grape